test/integration/dnspolicy_controller_test.go

```diff
@@ -171,21 +202,19 @@ var _ = Describe("DNSPolicy", Ordered, func() {
 		})
 
 		Context("weighted dnspolicy", func() {
-			dnsPolicy := &v1alpha1.DNSPolicy{}
+			var dnsPolicy *v1alpha1.DNSPolicy
 
 			BeforeEach(func() {
-
-				Eventually(func() bool {
-					if err := k8sClient.Get(ctx, client.ObjectKey{testNamespace, TestPlacedGatewayName}, dnsPolicy); err != nil {
-						return false
-					}
-					testModifyPolicyAddHealthCheckAndLoadBalancing(dnsPolicy)
-					if err := k8sClient.Update(ctx, dnsPolicy); err != nil {
+				dnsPolicy = testBuildDNSPolicyWithHealthCheck("test-dns-policy", TestPlacedGatewayName, testNamespace)
```

Comment:
@philbrookes I get what you were trying to do here, i had originally done something similar, but for whatever reason running these tests using `envtest` doesn't always appear to cause updates to happen as you would expect leading to flaky tests see https://github.com/Kuadrant/multicluster-gateway-controller/pull/321